src/utils/train.py

```diff
@@ -188,17 +194,24 @@ def evaluate_model(
     ground_truth = torch.cat(ground_truth).numpy()
     predictions = torch.cat(predictions).numpy()
 
-    average = "binary" if num_classes == 2 else "macro"
+    if no_average:
+        average = None
+    else:
+        average = "binary" if num_classes == 2 else "macro"
     precision, recall, f1_score, _ = precision_recall_fscore_support(ground_truth, predictions, average=average)
     accuracy = accuracy_score(ground_truth, predictions)
     result = {"accuracy": accuracy, "precision": precision, "recall": recall, "f1": f1_score}
 
     if group != "":
         result = {f"{group}/{k}": v for k, v in result.items()}
 
+    if no_average:
+        log_string = ",\n ".join(f"{k}: {v}" for k, v in result.items())
```

Comment:
I suggest to log table when no average (https://docs.wandb.ai/guides/data-vis/log-tables#create-tables)
There would be columns: class name, precision, recall, f1-score.

You can store class names inside the model or use range from 0 to `num_classes` if no class names available.